src/go/pt-galera-log-explainer/translate/translate.go

```diff
@@ -59,55 +59,78 @@ func GetDB() translationsDB {
 	return db
 }
 
+func (tu *translationUnit) UpdateTimestamp(ts time.Time) {
+	// we want to avoid gap of information, so the earliest proof should be kept
+	if tu.Timestamp.After(ts) {
+		tu.Timestamp = ts
+	}
+}
+
 func AddHashToIP(hash, ip string, ts time.Time) {
 	db.rwlock.Lock()
 	defer db.rwlock.Unlock()
-	db.HashToIP[hash] = translationUnit{Value: ip, Timestamp: ts}
+	latestValue, ok := db.HashToIP[hash]
+	if !ok {
+		db.HashToIP[hash] = &translationUnit{Value: ip, Timestamp: ts}
+	} else {
+		latestValue.UpdateTimestamp(ts)
+	}
```

Comment:
"it should not happen", but it's famous last words
Fixed, thanks

Reply:
Damn, it was evening, I was tired and I gave you wrong advise. Usually condition looks like `if ok` and fist code block checks when entity found in map, and in second block handled situation when it's not. I missed that you use inverted `if !ok` condition. 🤦‍♂️

Sorry, my bad. You code was correct, so I think you can revert the change.

Reply:
No worries! You were still right, a tiny "if latestvalue != nil"  won't hurt
I reversed the conditions to help readability, because `if ! ok || lastvalue == nil {` was really weird. `if ok && latestvalue != nil` is way more natural ofc